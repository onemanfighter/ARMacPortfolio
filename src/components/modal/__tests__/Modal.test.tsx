import { render, screen } from '@testing-library/react';
import ModalComponent from '../Modal';
import { ModalIDs } from '@store';
import { on } from 'events';

describe('Modal component', () => {
  const onModalClose = jest.fn();

  describe('search modal', () => {
    beforeEach(() => {
      onModalClose.mockReset();
    });

    it('should render correctly ', () => {
      const { container } = render(
        <ModalComponent
          isOpen={true}
          modalID={ModalIDs.SEARCH}
          onModalClose={onModalClose}
        />,
      );

      expect(container).toMatchSnapshot();
      expect(screen.getByText('Search')).toBeDefined();
    });

    it('should not render component if isOpen is false', () => {
      const { container } = render(
        <ModalComponent
          isOpen={false}
          modalID={ModalIDs.SEARCH}
          onModalClose={onModalClose}
        />,
      );

      expect(container).toMatchSnapshot();
      expect(screen.queryByText('Search')).toBeNull();
    });
  });
});
