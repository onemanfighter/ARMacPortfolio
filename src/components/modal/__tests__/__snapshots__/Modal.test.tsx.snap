// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component search modal should not render component if isOpen is false 1`] = `<div />`;

exports[`Modal component search modal should not render for none   1`] = `<div />`;

exports[`Modal component search modal should render correctly  1`] = `<div />`;
